src/nodeglpk.hpp

```diff
@@ -167,6 +176,35 @@ class EventEmitterDecorator : public ReentrantCWorker {
      Nan::AsyncWorker* decorated_;
 };
 
+class MemStatsDecorator : public Nan::AsyncWorker {
+ public:
+    MemStatsDecorator(Nan::AsyncWorker* decorated, std::shared_ptr<glp_memstats> memstats)
+        : Nan::AsyncWorker(nullptr), decorated_(decorated), memstats_(memstats) {}
+
+    virtual void HandleOKCallback() override { }
+    virtual void HandleErrorCallback() override { } 
+
+    virtual void WorkComplete() override {
+        MemStatsGuard mguard{memstats_};
+        decorated_->WorkComplete();
+        Nan::AsyncWorker::WorkComplete();
```

Comment:
doesn't the decorated do that already?

Reply:
decorated handles only decorated's business; but we have to also handle our own. for Nan::AsyncWorker#WorkComplete (as defined today) it is actually a NOP here, since callback is nullptr; but "just in case" it ever does cleanup work for the instance, I figured better to leave it in.